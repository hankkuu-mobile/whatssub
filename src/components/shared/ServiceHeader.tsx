import {
  Dimensions,
  Image,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  View } from 'react-native';
import React, { Component } from 'react';

import { IC_WRITE } from '../../utils/Icons';
import { IMG_MAIN } from '../../utils/Images';
import { colors } from '../../theme';
import styled from 'styled-components/native';

const Container = styled.View`
  width: 100%;
  background-color: ${colors.headerBackgroundLight};
  align-items: center;
  justify-content: center;
`;

const ImageContainer = styled.View`
  height: 86px;
  align-items: center;
  justify-content: center;
`;

const MainImage = styled.Image`
  max-height: 86px;
  width: ${Dimensions.get('window').width}px;
`;

const ServiceName = styled.Text`  
  font-family: 'spoqa-han-sans-bold';
  font-size: 16px;
`;

const ServiceNameInput = styled.TextInput`  
  font-family: 'spoqa-han-sans-bold';
  font-size: 16px;
`;

const ServiceDescription = styled.Text`  
  font-family: 'spoqa-han-sans-regular';
  font-size: 14px;
  color: ${colors.gray};
  margin-top: 13px;
  margin-bottom: 20px;
`;

interface Props {
  image?: number
  custom?: boolean
  serviceName?: string
  description?: string
  onChangeServiceName?(string): void
}

function Shared(props: Props) {
  const {
    image = IMG_MAIN,
    description = '서비스에 대한 요금과 결제정보를 입력해주세요.',
    custom = false,
    serviceName,
  } = props;
  const titleRef = React.useRef(null);
  return (
    <Container>
      <ImageContainer>
        <MainImage source={image} resizeMode='contain' />
      </ImageContainer>
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        {
          custom
            ? <ServiceNameInput
              ref={titleRef}
              placeholder='서비스명을 직접 입력해주세요.'
              value={serviceName}
              placeholderTextColor='black'
              onChangeText={props.onChangeServiceName}/>
            : <ServiceName>{serviceName}</ServiceName>
        }
        {
          custom
            ? <TouchableOpacity onPress={() => titleRef != null && titleRef.current.focus()}>
              <Image source={IC_WRITE} style={{ width: 14, height: 16, marginHorizontal: 4 }} />
            </TouchableOpacity>
            : null
        }
      </View>
      <ServiceDescription>{description}</ServiceDescription>
    </Container>
  );
}

export default Shared;
