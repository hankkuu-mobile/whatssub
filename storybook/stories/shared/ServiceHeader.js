import { IC_FACEBOOK } from '../../../src/utils/Icons';
import React from 'react';
import ServiceHeader from '../../../src/components/shared/ServiceHeader';
import { colors } from '../../../src/theme';
import { storiesOf } from '@storybook/react-native';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  justify-content: center;
  background-color: ${colors.gray};
`;

const Description = styled.Text`
  padding: 10px;
`;

const Story = () => {
  const [title, setTitle] = React.useState('');
  return (
    <Container>
      <Description>Default (custom) title : {title}</Description>
      <ServiceHeader
        custom
        onChangeServiceName={(text) => setTitle(text)}/>
      <Description>Casual mode</Description>
      <ServiceHeader
        serviceName='페이스북'
        image={IC_FACEBOOK}
        description='페이스북 구독 페이지'/>
    </Container>
  );
};

storiesOf('shared-ServiceHeader', module)
  .add('default', () => <Story/>)
;
