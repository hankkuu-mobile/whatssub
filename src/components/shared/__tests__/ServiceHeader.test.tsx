import 'react-native';
import * as React from 'react';
import { RenderResult, act, fireEvent, render } from '@testing-library/react-native';
import ServiceHeader from '../ServiceHeader';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[ServiceHeader] render', () => {
  beforeEach(() => {
    props = createTestProps({ });
    component = (
      <ServiceHeader {...props} />
    );
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      // const btn = testingLib.queryByTestId('btn');
      // act(() => {
      //   fireEvent.press(btn);
      // });
      // expect(cnt).toBe(3);
    });
  });
});
