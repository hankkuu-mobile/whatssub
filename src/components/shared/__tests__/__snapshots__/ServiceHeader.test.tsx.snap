// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ServiceHeader] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 86,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "maxHeight": 86,
            "width": 750,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "spoqa-han-sans-bold",
            "fontSize": 16,
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#99a2aa",
          "fontFamily": "spoqa-han-sans-regular",
          "fontSize": 14,
          "marginBottom": 20,
          "marginTop": 13,
        },
      ]
    }
  >
    서비스에 대한 요금과 결제정보를 입력해주세요.
  </Text>
</View>
`;
